src/Symfony/Component/HttpKernel/Kernel.php

```diff
@@ -683,7 +683,7 @@ protected function dumpContainer(ConfigCache $cache, ContainerBuilder $container
         $dumper = new PhpDumper($container);
 
         if (class_exists('ProxyManager\Configuration')) {
-            $dumper->setProxyDumper(new ProxyDumper());
+            $dumper->setProxyDumper(new ProxyDumper(md5((string) $cache)));
```

Comment:
This will be called at most twice in a common Symfony application flow, and that's when the cache is empty for the current environment and we call the `cache:clear` command.

The cache path changes between calls, which are (in absolute paths):
- In *dev* environment:
    - app/cache/dev/appDevDebugProjectContainer.php
    - app/cache/de_/ap_DevDebugProjectContainer.php
- In *prod* environment:
    - app/cache/prod/appProdProjectContainer.php
    - app/cache/pro_/ap_ProdProjectContainer.php

I think the risk of collision here is pretty much zero. Choosing `md5` was kind of arbitrary.

I made this PR just to make both executions different. In the future the cache path may not change between calls and we'll need to change the salt. Maybe a simple integer will do: start controlling how many containers the `Kernel` has dumped, and use this index as a salt (even un-hashed). I don't know if it pays off.